woocommerce/google-listings-and-ads
# Return settings prop as object in the REST API

### Changes proposed in this Pull Request:

Ref: p1724235667349019/1722414955.111429-slack-C02BB3F30TG

Reverts https://github.com/woocommerce/google-listings-and-ads/pull/2519

This PR fixes an error introduced in https://github.com/woocommerce/google-listings-and-ads/pull/2519 

At first, I understood that it was expected to get all empty props as an empty object. However, what was expected was only to get the Settings prop as an object.  Rest of the props should be as before.

This PR reverts https://github.com/woocommerce/google-listings-and-ads/pull/2519  and makes only settings to be returned as an object.

### Screenshots:

<img width="562" alt="Screenshot 2024-08-23 at 11 34 23" src="https://github.com/user-attachments/assets/f052531c-cb11-4180-a3ef-4b319bca167d">
<img width="319" alt="Screenshot 2024-08-23 at 11 34 46" src="https://github.com/user-attachments/assets/26f715a2-3db6-4552-93b6-e4454a023bac">
<img width="294" alt="Screenshot 2024-08-23 at 11 35 45" src="https://github.com/user-attachments/assets/4fa10e46-a983-48c2-ada6-df1dbd0eeb9b">
<img width="329" alt="Screenshot 2024-08-23 at 11 35 31" src="https://github.com/user-attachments/assets/d99ce190-6155-4465-bcb4-e00b51e5dccf">



### Detailed test instructions:


1.  Checkout this PR and 
2. Request `wp-json/wc/v3/products?gla_syncable=1` 
3. `attributes` field is shown as an array (when is empty and when is filled)
4. `gla_attributes` field is shown as an object (when is empty and when is filled)
5. ~~Request `wp-json/wc/v3/settings/general?gla_syncable=1`~~ 
6. ~~`settings` field is shown as an object (when is empty and when is filled)~~
5. Add the following filter which will return empty settings for the flat shipping rate:

```
add_filter(
	'woocommerce_shipping_instance_form_fields_flat_rate',
	function ( $defaults ) {
		return [];
	},
	10,
	1
);
```
6. Make a request to `v3/shipping/zones/YOUR_SHIPPING_ZONE/methods?gla_syncable=1` which includes a Flat Rate method.
7. See the response:

### Before:

![image](https://github.com/user-attachments/assets/eaa2c2ce-2c1e-49ca-b768-116e26618a86)


### After

![image](https://github.com/user-attachments/assets/e4d2ef15-1f43-4128-8205-d59927319353)


### Additional details:


### Changelog entry

>
